src/mpid/ch4/netmod/ofi/ofi_rma.h

```diff
@@ -1322,17 +1331,84 @@ static inline int MPIDI_NM_mpi_fetch_and_op(const void *origin_addr,
         goto fn_exit;
     }
 
-    /*  This can be optimized by directly calling the fi directly
-     *  and avoiding all the datatype processing of the full
-     *  MPID_Get_accumulate
-     */
-    mpi_errno = MPIDI_OFI_do_get_accumulate(origin_addr, 1, datatype,
-                                            result_addr, 1, datatype,
-                                            target_rank, target_disp, 1, datatype, op, win, NULL);
+    MPIDI_CH4U_RMA_OP_CHECK_SYNC(target_rank, win);
+    if (unlikely(target_rank == MPI_PROC_NULL))
+        goto fn_exit;
+
+    offset = target_disp * MPIDI_OFI_winfo_disp_unit(win, target_rank);
+
+    MPIDI_Datatype_check_size_lb(datatype, 1, bytes, true_lb);
```

Comment:
Just out of curiosity: in FOP, can `true_lb` be non-zero?

Reply:
Emm, should be always zero. Good catch

Reply:
done